plugins/vpp/ifplugin/vppcalls/vpp2001_379/dump_interface_vppcalls.go

```diff
@@ -921,6 +927,58 @@ func (h *InterfaceVppHandler) dumpGreDetails(interfaces map[uint32]*vppcalls.Int
 	return nil
 }
 
+// dumpGtpuDetails dumps GTP-U interface details from VPP and fills them into the provided interface map.
+func (h *InterfaceVppHandler) dumpGtpuDetails(interfaces map[uint32]*vppcalls.InterfaceDetails) error {
+	reqCtx := h.callsChannel.SendMultiRequest(&vpp_gtpu.GtpuTunnelDump{
+		SwIfIndex: ^uint32(0),
+	})
+	for {
+		gtpuDetails := &vpp_gtpu.GtpuTunnelDetails{}
+		stop, err := reqCtx.ReceiveReply(gtpuDetails)
+		if stop {
+			break // Break from the loop.
+		}
+		if err != nil {
+			return fmt.Errorf("failed to dump GTP-U tunnel interface details: %v", err)
+		}
+		_, ifIdxExists := interfaces[gtpuDetails.SwIfIndex]
+		if !ifIdxExists {
+			continue
+		}
+		// Multicast interface
+		var multicastIfName string
+		_, exists := interfaces[gtpuDetails.McastSwIfIndex]
+		if exists {
+			multicastIfName = interfaces[gtpuDetails.McastSwIfIndex].Interface.Name
+		}
+
+		if gtpuDetails.IsIPv6 == 1 {
+			interfaces[gtpuDetails.SwIfIndex].Interface.Link = &ifs.Interface_Gtpu{
```

Comment:
I'm somewhat uncomfortable with the amount of duplicated code in then-else branches.

Does it make sense to introduce a helper function for binary to text IP address conversion, like `IPString(gtpuDetails, gtpuDetails.SrcAddress)`?

Reply:
OK, let me refactor the if-else a bit.
Regarding the conversion function, this is what the net.IP module already does. Calling another function would not be too much shorted in my opinion.